File: backend/agents/state.py
from typing import TypedDict, List, Dict, Any, Optional

class AnalysisState(TypedDict):
    """
    TypedDict schema representing the shared state dictionary passed 
    between LangGraph nodes during repository multi-agent analysis.
    """
    repo_url: str
    owner: str
    repo_name: str
    default_branch: str
    file_paths: List[str]
    tree_raw: List[Dict[str, Any]]
    languages: Dict[str, int]
    contributors: List[Dict[str, Any]]
    tech_stack_data: Dict[str, Any]
    security_report: Dict[str, Any]
    
    # Outputs filled dynamically by agent nodes
    architecture_report: Optional[str]
    feature_report: Optional[str]
    resume_report: Optional[str]
    interview_report: Optional[str]
    recommendation_report: Optional[str]

File: backend/agents/state.js
export const analysisStateChannels = {
  repo_url: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  owner: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  repo_name: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  default_branch: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  file_paths: { value: (x, y) => (y !== undefined ? y : x), default: () => [] },
  tree_raw: { value: (x, y) => (y !== undefined ? y : x), default: () => [] },
  languages: { value: (x, y) => (y !== undefined ? y : x), default: () => ({}) },
  contributors: { value: (x, y) => (y !== undefined ? y : x), default: () => [] },
  tech_stack_data: { value: (x, y) => (y !== undefined ? y : x), default: () => ({}) },
  security_report: { value: (x, y) => (y !== undefined ? y : x), default: () => ({}) },
  
  architecture_report: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  feature_report: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  resume_report: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  interview_report: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
  recommendation_report: { value: (x, y) => (y !== undefined ? y : x), default: () => null },
};
